multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1346,6 +1412,37 @@ def _default_predict(
         strategy: str,
     ) -> List[Dict]:
 
+        if self._config.env.strategy == DEEPSPEED_OFFLOADING and DEEPSPEED_MODULE not in sys.modules:
+            # Need to initialize DeepSpeed and optimizer as currently required in Pytorch-Lighting integration of deepspeed.
+            from .optimization.deepspeed import CustomDeepSpeedStrategy
+
+            strategy = CustomDeepSpeedStrategy(
+                stage=3,
+                offload_optimizer=True,
+                offload_parameters=False,
+                allgather_bucket_size=self._config.env.deepspeed_allgather_size,
+                reduce_bucket_size=self._config.env.deepspeed_allreduce_size,
+            )
+            norm_param_names = get_norm_layer_param_names(self._model)
+            trainable_param_names = get_trainable_params_efficient_finetune(
+                norm_param_names, efficient_finetune=OmegaConf.select(self._config, "optimization.efficient_finetune")
+            )
+
+            optimization_kwargs = dict(
+                optim_type=self._config.optimization.optim_type,
+                lr_choice=self._config.optimization.lr_choice,
+                lr_schedule=self._config.optimization.lr_schedule,
+                lr=self._config.optimization.learning_rate,
+                lr_decay=self._config.optimization.lr_decay,
+                end_lr=self._config.optimization.end_lr,
+                lr_mult=self._config.optimization.lr_mult,
+                weight_decay=self._config.optimization.weight_decay,
+                warmup_steps=self._config.optimization.warmup_steps,
+            )
+        else:
+            optimization_kwargs = {}
+            trainable_param_names = []
```

Comment:
Do we still need this deepspeed strategy in inference? Isn't it only necessary for training?

Reply:
Consider adding a TODO to improve the design. Using `optimization_kargs` and `trainable_param_names` in inference is kind of confusing since no training and optimization are not expected in inference.

Reply:
Good point, added a TODO now. Unfortunately, specifying `optmization_kwargs` is currently a requirement in pytorch-lightning integration of deepspeed but should be removed as soon as possible. Calling deepspeed still needed in inference to run large models to reduce computation to `largest_layer_memory + 2 * params`.